include/aspect/mesh_deformation/fastscape.h

```diff
@@ -0,0 +1,725 @@
+       */
+      unsigned int bottom;
+
+      /**
+       * FastScape top boundary condition that determines topography at the FastScape top boundary.
+       * Where 1 represents a fixed height boundary (though this can still be uplifted through uplift velocities), and 0 a
+       * reflective boundary. When two opposing boundaries are reflective (e.g., top and bottom are both zero), then the boundaries
+       * become cyclic.
+       */
+      unsigned int top;
+
+      /**
+       * FastScape right boundary condition that determines topography at the FastScape right boundary.
+       * Where 1 represents a fixed height boundary (though this can still be uplifted through uplift velocities), and 0 a
+       * reflective boundary. When two opposing boundaries are reflective (e.g., left and right are both zero), then the boundaries
+       * become cyclic.
+       */
+      unsigned int right;
+
+      /**
+       * FastScape left boundary condition that determines topography at the FastScape left boundary.
+       * Where 1 represents a fixed height boundary (though this can still be uplifted through uplift velocities), and 0 a
+       * reflective boundary. When two opposing boundaries are reflective (e.g., left and right are both zero), then the boundaries
+       * become cyclic.
+       */
+      unsigned int left;
+
+      /**
+       * Paramters that set the fastscape boundaries periodic even though the ghost nodes are set 'fixed'
+       */
+      bool topbottom_ghost_nodes_periodic;
+      bool leftright_ghost_nodes_periodic;
+
+      /**
+       * Integer that holds the full boundary conditions sent to FastScape (e.g., 1111).
+       */
+      int bc;
+
+      /**
+       * Prescribed flux per unit length into the model through the bottom boundary (m^2/yr).
+       */
+      double bottom_flux;
+
+      /**
+       * Prescribed flux per unit length into the model through the top boundary (m^2/yr).
+       */
+      double top_flux;
+
+      /**
+       * Prescribed flux per unit length into the model through the right boundary (m^2/yr).
+       */
+      double right_flux;
+
+      /**
+       * Prescribed flux per unit length into the model through the left boundary (m^2/yr).
+       */
+      double left_flux;
+
+      // FastScape erosional parameters //
+      /**
+       * Drainage area exponent for the stream power law (m variable in FastScape surface equation).
+       */
+      double m;
+
+      /**
+       * Slope exponent for the steam power law (n variable in FastScape surface equation).
+       */
+      double n;
+
+      /**
+       * Slope exponent for multi-direction flow, where 0 is uniform, and 10 is steepest descent. (-1 varies with slope)
+       * (p variable in FastScape surface equation).
+       */
+      double p;
+
+      /**
+       * Bedrock deposition coefficient. Higher values deposit more sediment
+       * inside the domain. (G variable in FastScape surface equation).
+       */
+      double g;
```

Comment:
Could we use more descriptive names for these variables? Perhaps as follows?
```suggestion
      /**
       * Drainage area exponent for the stream power law. ($m$ variable in FastScape surface equation.)
       */
      double drainage_area_exponent_m;

      /**
       * Slope exponent for the steam power law. ($n$ variable in FastScape surface equation.)
       */
      double slope_exponent_n;

      /**
       * Slope exponent for multi-direction flow, where 0 is uniform, and 10 is steepest descent. (-1 varies with slope.)
       * ($p$ variable in FastScape surface equation.)
       */
      double slope_exponent_p;

      /**
       * Bedrock deposition coefficient. Higher values deposit more sediment
       * inside the domain. ($G$ variable in FastScape surface equation.)
       */
      double bedrock_deposition_g;
```